src/site/xdoc/userguide.xml

```diff
@@ -0,0 +1,214 @@
+        int inputOffset = 0;<br/>
+        int inputLen = input.length();<br/>
+        byte[] output = new byte[1024];<br/>
+        int outputOffset = 0;<br/>
+        //Initializes the cipher with ENCRYPT_MODE,key and iv.<br/>
+        cipher.init(Cipher.ENCRYPT_MODE, key, iv);<br/>
+        //Continues a multiple-part encryption/decryption operation for byte array.<br/>
+        cipher.update(input.getBytes("UTF-8"), inputOffset, inputLen, output, outputOffset);<br/>
+        //We should call do final at the end of encryption/decryption.<br/>
+        cipher.doFinal(inBuffer, outBuffer);<br/>
+        //Closes the cipher.<br/>
+        cipher.close();
+        </td>
+        </tr>
+        </table>
+
+
+        <h5>Usage of ByteBuffer Encryption/Decryption</h5>
+        <table>
+        <tr>
+        <td>
+        int bufferSize = 4096;<br/>
+        ByteBuffer inBuffer = ByteBuffer.allocateDirect(bufferSize);<br/>
+        ByteBuffer outBuffer = ByteBuffer.allocateDirect(bufferSize);<br/>
+        inBuffer.put("The data you want to encrypt or decrypt".getBytes("UTF-8"));<br/>
+        //Initializes the cipher with ENCRYPT_MODE,key and iv.<br/>
+        cipher..init(Cipher.ENCRYPT_MODE, key, iv);<br/>
+        //Continues a multiple-part encryption/decryption operation for byte array.<br/>
+        cipher.update(inBuffer, outBuffer);<br/>
+        //We should call do final at the end of encryption/decryption.<br/>
+        cipher.doFinal(inBuffer, outBuffer);<br/>
+        //Closes the cipher.<br/>
+        cipher.close();
+        </td>
+        </tr>
+        </table>
+
+
+        <h4>Usage of Stream API</h4>
+        <table>
+        <tr>
+        <td>
+        Properties properties = new Properties();<br/>
+        // Default value is com.intel.chimera.random.JavaSecureRandom.<br/>
+        properties.setProperty("chimera.crypto.secure.random.classes", "com.intel.chimera.random.OpensslSecureRandom,com.intel.chimera.random.OsSecureRandom");<br/>
+        //Gets the 'SecureRandom' instance.<br/>
+        SecureRandom secureRandom = SecureRandomFactory.getSecureRandom(properties);<br/>
+        // Available key length is: 16, 24, 32 bytes.<br/>
+        //Constructs the crypto key for cipher.<br/>
+        byte[] key = new byte[16];<br/>
+        //Generates random bytes and places them into the byte array.<br/>
+        secureRandom.nextBytes(key);<br/>
+        // IV length is always 16 bytes<br/>
+        //Constructs the crypto iv for cipher.<br/>
+        byte[] iv = new byte[16];<br/>
+        //Generates random bytes and places them into the byte array.<br/>
+        secureRandom.nextBytes(iv);<br/>
+        //Closes the SecureRandom.<br/>
+        secureRandom.close();<br/>
+        properties = new Properties();<br/>
+        // Default value is com.intel.chimera.crypto.JceCipher.<br/>
+        properties.setProperty("chimera.crypto.cipher.classes", "com.intel.chimera.crypto.OpensslCipher,com.intel.chimera.crypto.JceCipher");<br/>
```

Comment:
we can remove the code `Cipher cipher = Utils.getCipherInstance(CipherTransformation.AES_CTR_NOPADDING, properties);` to make it simple.